pinot-core/src/main/java/com/linkedin/pinot/core/indexsegment/generator/SegmentGeneratorConfig.java

```diff
@@ -433,30 +429,19 @@ public boolean isEnableStarTreeIndex() {
     return _enableStarTreeIndex;
   }
 
-  public void setEnableStarTreeIndex(boolean enableStarTreeIndex) {
-    _enableStarTreeIndex = enableStarTreeIndex;
-  }
-
-  public String getStarTreeIndexSpecFile() {
-    return _starTreeIndexSpecFile;
-  }
-
-  public void setStarTreeIndexSpecFile(String starTreeIndexSpecFile) {
-    _starTreeIndexSpecFile = starTreeIndexSpecFile;
-
-    // Setting the star tree index spec should automatically enable star tree generation, so that clients
-    // don't have to explicitly set both.
+  /**
+   * Enable star tree generation.
+   * @param starTreeIndexSpec Indexing spec for star tree. If null, then default values are used.
+   */
+  public void setEnableStarTreeIndex(StarTreeIndexSpec starTreeIndexSpec) {
```

Comment:
maybe change the name so that it doesn't sound like it should be a boolean field?

Reply:
How about enableStarTreeIndex()?

Reply:
i like setStarTreeIndexSpec or something without enable more, but as long as you feel that it isn't a boolean, fine with me

Reply:
I prefer to have 'enable' in the name because that is the primary function of this api. StarTreeIndexSpec can be null.